sway/desktop/output.c

```diff
@@ -651,4 +673,72 @@ void handle_new_output(struct wl_listener *listener, void *data) {
 	}
 
 	transaction_commit_dirty();
+
+	update_output_manager_config(server);
+}
+
+void handle_output_manager_apply(struct wl_listener *listener, void *data) {
+	struct sway_server *server =
+		wl_container_of(listener, server, output_manager_apply);
+	struct wlr_output_configuration_v1 *config = data;
+
+	struct wlr_output_configuration_head_v1 *config_head;
+	// First disable outputs we need to disable
+	wl_list_for_each(config_head, &config->heads, link) {
+		struct wlr_output *wlr_output = config_head->state.output;
+		struct sway_output *output = wlr_output->data;
+		if (!output->enabled || config_head->state.enabled) {
+			continue;
+		}
+		struct output_config *oc = new_output_config("temp");
+		oc->enabled = false;
+		// There is no explicit way to set DPMS through the protocol.
+		// Users probably expect that disabling a monitor will turn it off.
+		oc->dpms_state = DPMS_OFF;
```

Comment:
No need to touch DPMS, setting `enabled = false` is enough. DPMS is deliberately left out of the protocol.

Reply:
Oh, right, was reading the wrong code path in apply_output_config